tests/protocol/test_wire.py

```diff
@@ -1256,24 +1254,22 @@ def assert_no_exception(test_case, test_function, expected_error):
 
         def test_error_in_http_request(test_case, mock_response, expected_error):
             def do_mock_request():
-                with mock_wire_protocol(mockwiredata.DATA_FILE_VM_SETTINGS) as protocol:
-                    protocol.do_not_mock = lambda method, url: method == "GET" and self.is_host_plugin_vm_settings_request(url)
-
-                    def http_get_vm_settings(_method, _host, _relative_url, **_):
+                def http_get_handler(url, *_, **__):
+                    if self.is_host_plugin_vm_settings_request(url):
                         if isinstance(mock_response, Exception):
                             raise mock_response
                         return mock_response
+                    return None
 
-                    with patch("azurelinuxagent.common.utils.restutil._http_request", side_effect=http_get_vm_settings):
```

Comment:
I was mocking the internal method to bypass the error handling done by the public method, but that is no longer needed.